File: Practica 1 Auxiliatura/Introduccion a poo/Eje2/Bus.py
class Bus:
    def __init__(self, capacidad):
        self.capacidad = capacidad
        self.pasajeros = 0
        self.costo_pasaje = 1.50

    def subir_pasajeros(self, cantidad):
        if self.pasajeros + cantidad <= self.capacidad:
            self.pasajeros += cantidad
            print(f"{cantidad} pasajeros subieron al bus.")
        else:
            disponibles = self.capacidad - self.pasajeros
            print(f"Solo pueden subir {disponibles} pasajeros. Bus lleno.")
            self.pasajeros = self.capacidad

    def cobrar_pasaje(self):
        return self.pasajeros * self.costo_pasaje

    def asientos_disponibles(self):
        return self.capacidad - self.pasajeros

    def mostrar_estado(self):
        print(f"Pasajeros actuales: {self.pasajeros}")
        print(f"Asientos disponibles: {self.asientos_disponibles()}")
        print(f"Total recaudado: Bs. {self.cobrar_pasaje():.2f}")


# main
if __name__ == "__main__":
    mi_bus = Bus(40)

    mi_bus.subir_pasajeros(15)
    mi_bus.mostrar_estado()

    mi_bus.subir_pasajeros(30)
    mi_bus.mostrar_estado()

File: Practica 1 Auxiliatura/Introduccion a poo/Eje2/Bus.java
package Eje2;

public class Bus {
    private int capacidad;
    private int pasajeros;
    private final double costoPasaje = 1.50;

    public Bus(int capacidad) {
        this.capacidad = capacidad;
        this.pasajeros = 0;
    }

    public void subirPasajeros(int cantidad) {
        if (pasajeros + cantidad <= capacidad) {
            pasajeros += cantidad;
            System.out.println(cantidad + " pasajeros subieron al bus.");
        } else {
            int disponibles = capacidad - pasajeros;
            System.out.println("Solo pueden subir " + disponibles + " pasajeros. Bus lleno.");
            pasajeros = capacidad;
        }
    }

    public double cobrarPasaje() {
        return pasajeros * costoPasaje;
    }

    public int asientosDisponibles() {
        return capacidad - pasajeros;
    }

    public void mostrarEstado() {
        System.out.println("Pasajeros actuales: " + pasajeros);
        System.out.println("Asientos disponibles: " + asientosDisponibles());
        System.out.println("Total recaudado: Bs. " + cobrarPasaje());
    }
}